modules/gdnative/nativescript/nativescript.cpp

```diff
@@ -1396,6 +1420,22 @@ void NativeScriptLanguage::thread_exit() {
 
 #endif // NO_THREADS
 
+bool NativeScriptLanguage::handles_global_class_type(const String &p_type) const {
+	return p_type == "NativeScript";
+}
+
+String NativeScriptLanguage::get_global_class_name(const String &p_path, String *r_base_type, String *r_icon_path) const {
+	Ref<NativeScript> script = ResourceLoader::load(p_path, "NativeScript");
```

Comment:
This actually loads and initializes the library. That's the only way you can get information like "base type", but it's still suboptimal.

This is just a comment, you don't need to change anything here, it's fine, just realized that it's a bit "meh" to load and unload libraries all the time when all you need is three strings :smile:

Reply:
Is there perhaps a better way we should be defining / accessing these values then? Perhaps some way to directly access the data contained within the resource file itself? I mean, I could directly open up the file and parse it for the information...but that's kinda hacky.

Reply:
Actually, if it inherits Resource, is there a way to load it AS a Resource, rather than as a NativeScript, and then just directly access the member properties on it using .get()?

Reply:
I don't think so, if you just parse it as a regular resource it will dispatch to the custom code anyway AFAIK. And yeah, there's information like "base type" which you only know at runtime, unless we make the user specify it in the `.gdns` file as well (would only be necessary for global classes then)